src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2692,3 +2725,319 @@ map<long, long> JasmineGraphInstanceService::calculateLocalOutDegreeDistribution
         }
     }
 }
+
+bool duplicateCentralStore(int thisWorkerPort, int graphID, int partitionID,
+                     std::vector<string> workerSockets, std::string masterIP) {
+
+    Utils utils;
+    std::string aggregatorFilePath = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.aggregatefolder");
+    std::string dataFilePath = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+
+    std::string centralGraphIdentifierUnCompressed = to_string(graphID) +"_centralstore_"+ to_string(partitionID);
+    std::string centralStoreFileUnCompressed = dataFilePath + "/" + centralGraphIdentifierUnCompressed;
+    std::string centralStoreFileUnCompressedDestination = aggregatorFilePath + "/" + centralGraphIdentifierUnCompressed;
+
+    // temporary copy the central store into the aggregate folder in order to compress and send
+    utils.copyFile(centralStoreFileUnCompressed, aggregatorFilePath);
+
+    // compress the central store file before sending
+    utils.compressFile(centralStoreFileUnCompressedDestination);
+
+    std::string centralGraphIdentifier = to_string(graphID) +"_centralstore_"+ to_string(partitionID) + ".gz";
+    std::string centralStoreFile = aggregatorFilePath + "/" + centralGraphIdentifier;
+    instance_logger.log("###INSTANCE### centralstore " + centralStoreFile,"info");
+
+    for (vector<string>::iterator workerIt=workerSockets.begin(); workerIt!=workerSockets.end(); ++workerIt) {
+
+        std::vector <string> workerSocketPair;
+        boost::split(workerSocketPair, *workerIt, boost::is_any_of(":"));
+        struct stat s;
+        if (stat(centralStoreFile.c_str(),&s) == 0) {
+            if (s.st_mode & S_IFREG) {
+
+                string host = workerSocketPair[0];
```

Comment:
Is there a change of `workerSocketPair` becoming shorter than 4 items? Shall we check that in advance using if check?

Reply:
added a validation to check the size

Reply:
Ok

Reply:
But then why do we continue continue processing from there onwards? We should return?

Reply:
Added a return

Reply:
Ok